prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -5374,6 +5411,62 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
             .min()
             .unwrap()
     }
+
+    /// A local version of encode_place
+    fn encode_place(
+        &self,
+        place: &mir::Place<'tcx>,
+    ) -> EncodingResult<(vir::Expr, Vec<vir::Stmt>, ty::Ty<'tcx>, Option<usize>)> {
+        let (encoded_place, ty, variant_idx) = self.mir_encoder.encode_place(place)?;
+        // TODO: actual encoding of e.g. array access here
+        let (encoded_expr, encoding_stmts) = self.postprocess_place_encoding(encoded_place)?;
+
+        Ok((encoded_expr, encoding_stmts, ty, variant_idx))
+    }
+
+    fn encode_projection(
+        &self,
+        local: mir::Local,
+        projection: &[mir::PlaceElem<'tcx>],
+    ) -> EncodingResult<(vir::Expr, Vec<vir::Stmt>, ty::Ty<'tcx>, Option<usize>)> {
+        let (encoded_place, ty, variant_idx) = self.mir_encoder.encode_projection(local, projection)?;
+        let (encoded_expr, encoding_stmts) = self.postprocess_place_encoding(encoded_place)?;
+
+        Ok((encoded_expr, encoding_stmts, ty, variant_idx))
+    }
+
+    fn encode_deref(
+        &self,
+        encoded_base: PlaceEncoding,
+        base_ty: ty::Ty<'tcx>,
+    ) -> EncodingResult<(vir::Expr, Vec<vir::Stmt>, ty::Ty<'tcx>, Option<usize>)> {
+        let (encoded_place, ty, variant_idx) = self.mir_encoder.encode_deref(encoded_base, base_ty)?;
+        let (encoded_expr, encoding_stmts) = self.postprocess_place_encoding(encoded_place)?;
+
+        Ok((encoded_expr, encoding_stmts, ty, variant_idx))
+    }
```

Comment:
Just looking at the code in this PR, I think that this function is not needed. The only usage seems to be `self.encode_deref(PlaceEncoding::Expr(..), ..)` and `MirEncoder::encode_deref` only returns `PlaceEncoding::Expr(..)`. Do you foresee some new use cases?

Reply:
removed it, as discussed in the previous comment thread.